src/controllers/__tests__/request.test.js

```diff
@@ -0,0 +1,278 @@
+const request = require('supertest');
+const server = require('../../app');
+const { closeDatabase, clearDatabase } = require('../../db/connection');
+
+let ownerId;
+let authCookie;
+let RequestId;
+const nonExistingRequestId = '5e9f8f9b9b8f8e0e8c8b4567';
+const invalidRequestId = 'invalidId123';
+const trueRequest = {
+  name: 'Sofa',
+  description: 'A comfortable and medium size sofa',
+  owner: ownerId,
+  type: 'Stationery',
+};
+const trueRequest2 = {
+  name: 'Sofa2',
+  description: 'A comfortable and medium size sofa',
+  owner: ownerId,
+  type: 'Stationery',
+};
+const invalidTypeRequest = {
+  name: 'Sofa',
+  description: 'A comfortable and medium size sofa',
+  type: 'invalid',
+};
+const noNameRequest = {
+  description: 'A comfortable and medium size sofa',
+  owner: ownerId,
+  type: 'Stationery',
+};
+const noDescriptionRequest = {
+  name: 'Sofa',
+  owner: ownerId,
+  type: 'Stationery',
+};
+const invalidPhotoRequest = {
+  name: 'Sofa',
+  description: 'A comfortable and medium size sofa',
+  owner: ownerId,
+  type: 'Stationery',
+  photo: 'invalidImage',
+};
+const mockUser = {
+  username: 'mock.user',
+  firstName: 'mock',
+  lastName: 'user',
+  email: 'mockuser@gmail.com',
+  password: 'Password1234',
+  passwordConfirm: 'Password1234',
+  address: 'Central Perk, New York',
+  acceptTerms: true,
+};
+
+beforeAll(async () => {
+  await clearDatabase();
+  const signUpResponse = await request(server)
+    .post('/api/auth/signup')
+    .set('Content-Type', 'application/json')
+    .send(mockUser);
+  ownerId = signUpResponse.body._id;
+  trueRequest.owner = ownerId;
+  trueRequest2.owner = ownerId;
+  noNameRequest.owner = ownerId;
+  noDescriptionRequest.owner = ownerId;
+  invalidPhotoRequest.owner = ownerId;
+  const signInResponse = await request(server)
+    .post(`/api/auth/signin`)
+    .set('Content-Type', 'application/json')
+    .send({ username: mockUser.username, password: mockUser.password });
+  [authCookie] = signInResponse.headers['set-cookie'];
+});
+
+afterAll(async () => {
+  await clearDatabase();
+  await closeDatabase();
+  server.close();
+});
+
+describe('Request Endpoints', () => {
+  describe('GET /api/requests/', () => {
+    test('Should response with an error message when there are no requests', async () => {
+      await clearDatabase();
+      const response = await request(server).get('/api/requests/');
```

Comment:
You don't need to clear the database here since this is the first test and you are already clearing it in the `beforeAll` function

Reply:
Actually, this test was written after another one, but then I moved it here and forget to delete ` await clearDatabase();`